pacfinder_darwin.go

```diff
@@ -20,104 +19,122 @@ package main
 #include <SystemConfiguration/SystemConfiguration.h>
 
 static SCDynamicStoreRef SCDynamicStoreCreate_trampoline() {
-	return SCDynamicStoreCreate(kCFAllocatorDefault, CFSTR("alpaca"), NULL, NULL);
+    return SCDynamicStoreCreate(kCFAllocatorDefault, CFSTR("alpaca"), NULL, NULL);
 }
 
 typedef const CFStringRef CFStringRef_Const;
-
-const CFStringRef_Const kProxiesSettings = CFSTR("State:/Network/Global/Proxies");
-const CFStringRef_Const kProxiesAutoConfigURLString = CFSTR("ProxyAutoConfigURLString");
-const CFStringRef_Const kProxiesProxyAutoConfigEnable = CFSTR("ProxyAutoConfigEnable");
-
 */
 import "C"
 import (
+	"log"
 	"unsafe"
 )
 
 type pacFinder struct {
 	pacUrl   string
 	storeRef C.SCDynamicStoreRef
+	auto     bool
 }
 
 func newPacFinder(pacUrl string) *pacFinder {
 	if pacUrl != "" {
-		return &pacFinder{pacUrl, 0}
+		return &pacFinder{pacUrl, 0, false}
 	}
-
-	return &pacFinder{"", C.SCDynamicStoreCreate_trampoline()}
-}
-
-func (finder *pacFinder) findPACURL() (string, error) {
-	if finder.storeRef == 0 {
-		return finder.pacUrl, nil
+	storeRef := C.SCDynamicStoreCreate_trampoline()
+	if storeRef == 0 {
+		log.Fatalf("Failed to create SCDynamicStore")
 	}
-
-	//start := time.Now()
-	url := finder.getPACUrl()
-
-	//elapsed := time.Since(start)
-	//log.Printf("PacUrl found in %v", elapsed)
-
-	return url, nil
+	return &pacFinder{"", storeRef, true}
 }
 
 func (finder *pacFinder) pacChanged() bool {
 	if url, _ := finder.findPACURL(); finder.pacUrl != url {
 		finder.pacUrl = url
 		return true
 	}
-
 	return false
 }
 
-func (finder *pacFinder) getPACUrl() string {
-	dict := C.CFDictionaryRef(C.SCDynamicStoreCopyValue(finder.storeRef, C.kProxiesSettings))
+func (finder *pacFinder) findPACURL() (string, error) {
+	if finder.storeRef == 0 {
+		return finder.pacUrl, nil
+	}
 
-	if dict == 0 {
+	// Using SCDynamicStoreCopyProxies to get PAC URL
+	pacUrl := finder.getPACUrlFromSCDynamicStoreCopyProxies()
+	if pacUrl != "" {
+		// log.Printf("Using PAC URL from SCDynamicStoreCopyProxies method: %s", pacUrl)
+		return pacUrl, nil
+	}
+
+	// log.Printf("No PAC URL found using SCDynamicStoreCopyProxies")
+	return "", nil
+}
+
+func (finder *pacFinder) getPACUrlFromSCDynamicStoreCopyProxies() string {
```

Comment:
I think you can delete this and inline getPACUrlFromSCDynamicStoreCopyProxies() into findPACURL(). Now that you're not using two separate methods to retrieve the PAC URL, it seems unnecessary to have this code across two functions, and anyway the (commented out) logs are redundant with the logs from pacfetcher.go that I mentioned in the other comment.

```suggestion
```